src/SmartlingApi.php

```diff
@@ -122,19 +112,32 @@ protected function sendRequest($uri, $requestData, $method)
             }
         }
 
-        $guzzle_response = $this->httpClient->request($method, $this->baseUrl . '/' . $uri, $options);
+        // Avoid double slashes in final URL.
+        $uri = ltrim($uri, "/");
 
+        $guzzle_response = $this->httpClient->request($method, $this->baseUrl . $uri, $options);
         $response_body = (string) $guzzle_response->getBody();
-        if (strpos($response_body, '<?xml') === 0) {
+
+        // Catch all errors from Smartling and throw appropriate exception.
+        if ($guzzle_response->getStatusCode() >= 400) {
+            $error_response = json_decode($response_body, TRUE);
+
+            if (!$error_response || empty($error_response['response']['messages'])) {
+                throw new SmartlingApiException('Bad response format from Smartling');
+            }
+
+            throw new SmartlingApiException(implode(' || ', $error_response['response']['messages']), $guzzle_response->getStatusCode());
+        }
+
+        // "Download file" method return translated file directly.
+        if ('file/get' == $uri) {
```

Comment:
if is a good practice to use "===" instead of "==", unless there is a really good reason not to do it.